tensorboard/webapp/widgets/data_table/data_table_component.ts

```diff
@@ -223,41 +228,50 @@ export class DataTableComponent implements OnDestroy {
     this.sortDataBy.emit({header, order: SortingOrder.ASCENDING});
   }
 
-  dragStart(header: ColumnHeaders) {
-    this.draggingHeader = header;
+  dragStart(header: ColumnHeader) {
+    this.draggingHeaderType = header.type;
 
     // This stop the end drag animation
     document.addEventListener('dragover', preventDefault);
   }
 
   dragEnd() {
-    if (!this.draggingHeader || !this.highlightedColumn) {
+    if (!this.draggingHeaderType || !this.highlightedColumnType) {
       return;
     }
 
     this.orderColumns.emit(
       this.moveHeader(
-        this.headers.indexOf(this.draggingHeader),
-        this.headers.indexOf(this.highlightedColumn)
+        this.headers.findIndex((element) =>
+          isHeaderOfType(this.draggingHeaderType!, element)
```

Comment:
This function seems unnecessarily complicated. Could you either wrap the entire call with something along the lines of `getIndexOfDraggingHeader` or else remove the `isHeaderOfType` function?

Reply:
Good suggestion. Done.